Shared-Layer/ApiServices/CustomAuthStateProvider.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System.IdentityModel.Tokens.Jwt;
+using System.Net.Http.Headers;
+using System.Security.Claims;
+using Blazored.LocalStorage;
+using Microsoft.AspNetCore.Components.Authorization;
+
+namespace Shared_Layer.ApiServices
+{
+    public class CustomAuthStateProvider : AuthenticationStateProvider
+    {
+        private readonly ILocalStorageService _localStorageService;
+        private readonly HttpClient _httpClient;
+
+        public CustomAuthStateProvider(ILocalStorageService localStorageService, HttpClient httpClient)
+        {
+            _localStorageService = localStorageService;
+            _httpClient = httpClient;
+        }
+
+        public override async Task<AuthenticationState> GetAuthenticationStateAsync()
+        {
+            var token = await _localStorageService.GetItemAsync<string>("authToken");
+
+            if (string.IsNullOrEmpty(token))
+            {
+                return new AuthenticationState(new ClaimsPrincipal(new ClaimsIdentity()));
+            }
+
+            _httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", token);
```

Comment:
move this line of code to be placed like this.
 _httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", token);
 return new AuthenticationState(user)